app/src/main/java/com/futurice/freesound/utils/TimeSkipScheduler.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2016 Futurice GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.futurice.freesound.utils;
+
+import android.support.annotation.NonNull;
+
+import java.util.concurrent.TimeUnit;
+
+import io.reactivex.Scheduler;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.disposables.Disposables;
+
+/**
+ * Scheduler that executes immediately even the time based actions.
+ */
+public final class TimeSkipScheduler extends Scheduler {
```

Comment:
The thing is that until we have a separate module `testcommon`, we would not be able to use it in instrumentation tests as it wouldn't be visible for that package